### PR TITLE
Update the library stabilization guide to refer to the new placeholder system

### PR DESCRIPTION
**Pull request by est31**
Due to https://github.com/rust-lang/rust/pull/100591 the stabilization process is changed to use `CURRENT_RUSTC_VERSION` instead of spelling out whatever version is currently the latest. This placeholder will be replaced during the release process with the correct version.

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]
thank you for pushing this through :heart: changes to the guide seem fine, but I think we should hold off on merging them until your PR is merged in rust-lang/rust.

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**
